### PR TITLE
Support for cloud mode in Google Analytics 4

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Updated doc to accommodate cloud mode details

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: destinations/analytics/google-analytics-4/setting-up-google-analytics-4-in-rudderstack.mdx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+<div class="infoBlock">
+  
+ Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain the API Secret and Measurement Id.
+  </div>
+
+- **Firebase App Id**: Enter the Firebase App ID which is the identifier for Firebase app. It can be found in Firebase dashboard under **Project Settings > General**.
+
+<div class="infoBlock">
+  
+ If you are using the <code class="inline-code">gtag</code>, <strong>API Secret</strong> and <strong>Measurement Id</strong> are the required fields. However, if you are using the Firebase app, <strong>Firebase App Id</strong> and <strong>API Secret</strong> are the required fields.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/setting-up-google-analytics-4-in-rudderstack.mdx:45]:
```suggestion
 If you are using the <code class="inline-code">gtag</code>, <strong>API Secret</strong> and <strong>Measurement Id</strong> are the required fields. However, if you are using the <code class="inline-code">firebase</code> client, <strong>Firebase App Id</strong> and <strong>API Secret</strong> are the required fields.
```

File: destinations/analytics/google-analytics-4/setting-up-google-analytics-4-in-rudderstack.mdx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  - If enabled, user ID is set to the identified visitors and sent as explained in the [Cookies and user identification](https://developers.google.com/tag-platform/devguides/cookies#set_user_id) guide.
+  - If disabled, userId is not sent and Rudderstack only sends the user traits set as `user_properties` to Google Analytics. The user properties are set on the `gtag` client as explained in the [Set user properties](https://developers.google.com/analytics/devguides/collection/ga4/user-properties?technology=websites) guide.
+
+### Client-side events filtering
+
+<img src="/assets/event-stream-destinations/google-analytics-4-connection-settings2.png" alt="Google Analytics 4 Connection Settings in RudderStack"/>
+
+This option is applicable only if you're sending events to Google Analytics 4 via the web device mode. Refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/event-filtering/">Client-side Event Filtering</a> guide for more information on this feature.
+
+### Web SDK settings
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/setting-up-google-analytics-4-in-rudderstack.mdx:68]: OneTrust Cookie Category is missing from connection settings image

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    location_id: "L_12345"
+  }]
+});
+```
+
+## E-commerce
+
+RudderStack supports e-commerce tracking for Google Analytics 4. You can use the [RudderStack e-commerce specification](https://rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-ecommerce-events-specification/) for sending events while instrumenting your site with the RudderStack SDK.
+
+The below table lists the mappings of the `track` events and properties that are passed to Google Analytics 4 e-commerce events and properties:
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:59]:
```suggestion
The below table lists the mappings of the `track` events and properties that are passed to Google Analytics 4 events and properties:
```
GA4 docs have not mentioned about ecommerce event [Ref](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference/events)

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+The <strong>Payment Info Entered</strong> event is mapped on the basis of parameters passed to it:
+<ul>
+  <li>If <code class=“inline-code”>payment_method</code> is passed along with the other fields, it maps to <code class=“inline-code”>add_payment_info</code></li> event.
+ <li>If <code class=“inline-code”>shipping_method</code> is passed along with the other fields, it maps to <code class=“inline-code”>add_shipping_info</code></li> event.
+  </ul>
+</div>
+
+<div class="infoBlock">
+
+There must be an <code class="inline-code">id</code> and <code class="inline-code">name</code> associated for each product in the order. Refer to the <a href="https://developers.google.com/gtagjs/reference/ga4-events#view_item_list">Item list</a> for more information on Google Analytics 4 e-commerce events and corresponding properties.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:153]:
```suggestion
There must be an <code class="inline-code">id</code> and <code class="inline-code">name</code> associated for each product in the order. Refer to the <a href="https://developers.google.com/gtagjs/reference/ga4-events#view_item_list">Item list</a> for more information on Google Analytics 4 events and corresponding properties.
```
The item list referenced in the url is specific to view_item_list event.

Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:153]: The statement is quite unclear. Do we mean `item_id` and `item_name` must be present inside Items parameters

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+| RudderStack    | Google Analytics 4  |
+| :-------------| :-----------------|
+| product_id         | `item_id`        |
+| name               | `item_name`      |
+| price              | `price`          |
+| position           | `index`          |
+| category           | `item_category`  |
+| list_id            | `item_list_id`   |
+| category           | `item_list_name` |
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:166]: There are a few more mapping. Let's update this

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+
+
+### Rules for naming events
+
+You must follow the below rules while choosing a name for the new events:
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:172]: This applies to custom events only. [Ref](https://support.google.com/analytics/answer/10085872#zippy=%2Creserved-prefixes-and-event-names%2Cweb%2Cin-this-article)

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx (hunk 4 above)
Comment (akashrpo) [reply destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx]: @utsabc any suggestion here. This statement is part of our device mode doc.

File: destinations/analytics/google-analytics-4/index.mdx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+# slug: "/docs/destinations/analytics/google-analytics-4"
+title: "Google Analytics 4"
+description: Step-by-step guide to send your event data from RudderStack to Google Analytics 4.
+---
+
+# Google Analytics 4
+
+[Google Analytics 4](https://analytics.google.com/) is an analytics service that enables you to measure traffic and engagement across your websites and apps. It uses machine learning to reveal useful insights about the customer journey across different platforms and devices. 
+
+RudderStack supports the `gtag` way of tagging in websites. Google Analytics 4 can be used for websites, apps, or both together - whereas Universal Analytics properties only support websites.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/index.mdx:11]: We support for gtag and firebase both depending on connection modes.

Comment (akashrpo) [reply destinations/analytics/google-analytics-4/index.mdx:11]: Maybe we can just mention a small description of GA4 here and its details can be mentioned in their respective section

Comment (ishachopra0401) [reply destinations/analytics/google-analytics-4/index.mdx]: done

File: destinations/analytics/google-analytics-4/index.mdx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        </g>
+      </svg>
+    </span>
+    <span class="mainRefText">
+      Setting Up Google Analytics 4 in RudderStack
+    </span>
+  </div>
+</a>
+
+<a class="pageRef" href="google-analytics-4-device-mode/">
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/index.mdx:45]: Let's put cloud mode first

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+  </tr>
+  <tr>
+    <td>Promotion Viewed</br>Promotion Clicked</td>
+    <td><code class=“inline-code”>view_promotion</code></br><code class=“inline-code”>select_promotion</code></td>
+    <td><code class=“inline-code”>position</code></td>
+    <td><code class=“inline-code”>location_id</code></td>
+  </tr>
+   <tr>
+    <td>Product Viewed</br>Product Added</br>Product Removed</br>Cart Viewed</br>Checkout Started</br>Product Added to Wishlist</td>
+    <td><code class=“inline-code”>view_item</code></br><code class=“inline-code”>add_cart</code></br><code class=“inline-code”>remove_from_cart</code></br><code class=“inline-code”>view_cart</code></br><code class=“inline-code”>begin_checkout</code></br><code class=“inline-code”>add_to_wishlist</code></td>
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:96]:
```suggestion
    <td><code class=“inline-code”>view_item</code></br><code class=“inline-code”>add_to_cart</code></br><code class=“inline-code”>remove_from_cart</code></br><code class=“inline-code”>view_cart</code></br><code class=“inline-code”>begin_checkout</code></br><code class=“inline-code”>add_to_wishlist</code></td>
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+### What might be the reason for an error message from Google Analytics?
+
+Google Analytics 4 has some reserved prefixes and events names which cannot be used. The list of such prefixes is mentioned below:
+
+- _ (underscore)
+- firebase_
+- ga_
+- google_
+- gtag.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:184]: These are for custom events

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+For the list of reserved event names, refer to the [Reserved Names](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#reserved_names) guide.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:186]: This link provided here are for GA4 standard events

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx (hunk 10 above)
Comment (akashrpo) [reply destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:184]: We can provide [this](https://support.google.com/analytics/answer/10085872#zippy=%2Creserved-prefixes-and-event-names%2Cweb) link for custom events.

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+- Event names are case sensitive. For example, `my_event` and `My_Event` are two distinct events.
+- Event names must start with a letter. Only letters, numbers, and underscores are permitted. **DO NOT** use spaces.
+- Do not use reserved prefixes and event names. Refer to the <a href="#faq">FAQ</a> section to know more about the reserved event names and prefixes.
+
+## FAQ
+
+### What might be the reason for an error message from Google Analytics?
+
+Google Analytics 4 has some reserved prefixes and events names which cannot be used. The list of such prefixes is mentioned below:
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:178]: Lets seperate this

**Comment by utsabc** [comment]
Can we create a complete mapping table for all the properties so that user don't have to visit ga4 api docs
@akashrpo

File: destinations/analytics/google-analytics-4/google-analytics-4-device-mode.mdx [hunk 13]
```diff
@@ -0,0 +1,114 @@
+---
+# slug: "/docs/destinations/analytics/google-analytics-4-device-mode"
+title: "Device Mode"
```
Comment (utsabc) [inline destinations/analytics/google-analytics-4/google-analytics-4-device-mode.mdx:3]: For device mode we have couple of UI configs @ishachopra0401 can we please explain those in docs
@akashrpo please help Isha on this

File: destinations/analytics/google-analytics-4/google-analytics-4-device-mode.mdx [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+description: Detailed technical documentation on sending events to Google Analytics 4 using the RudderStack Web Device mode.
+---
+
```
Comment (utsabc) [inline destinations/analytics/google-analytics-4/google-analytics-4-device-mode.mdx:6]: Also lets mention that the user has select client as gtag and enable native sdk for device mode

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx (hunk 4 above)
Comment (utsabc) [reply destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:145]: Do we support items array or product array. If we are mapping from product array we should mention that

Comment (utsabc) [reply destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:145]: We should be clear about what structure we are expecting from user and to what structure we are mapping

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+| Product Viewed              | Required      |
+| Product Added              | Required         |
+| Product removed           | Required          |
+| Cart Viewed           | Required  |
+| Checkout Started           | Required  |
+| Payment Info entered           | Required  |
+| Order Completed           | Required  |
+| Order Refunded           | Optional  |
+| Product Added to Wishlist           | Required  |
+| View Search Result           | Optional  |
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:178]:
```suggestion
| View Search Results           | Optional  |
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+| variant           | `item_variant`  |
+| quantity           | `quantity`  |
+
+
+The below mentioned e-commerce events include [`items`](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference/events#view_item_list) parameter which accepts an `Item` array. 
+
+| RudderStack    | Presence of `items` parameter |
+| :-------------| :-----------------|
+| Product List Viewed         | Required        |
+| Product Clicked         | Optional      |
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:168]:
```suggestion
| Product Clicked         | Required      |
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 17]
```diff
@@ -0,0 +1,339 @@
...
+
+The reserved prefixes in Google Analytics 4 cannot be used for custom events. The list of such prefixes is mentioned below:
+
+- _ (underscore)
+- firebase_
+- ga_
+- google_
+- gtag.
+
+Refer to the [Google document](https://support.google.com/analytics/answer/10085872#zippy=%2Cweb%2Cin-this-article%2Creserved-prefixes-and-event-names) for more information.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:330]:
```suggestion
Refer to the [GA4 document](https://support.google.com/analytics/answer/10085872#zippy=%2Cweb%2Cin-this-article%2Creserved-prefixes-and-event-names) for more information.
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 18]
```diff
@@ -0,0 +1,339 @@
...
+
+### What are the reserved event names in Google Analytics 4?
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:332]:
```suggestion
### What are the reserved event, parameter, user property names in Google Analytics 4?
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 19]
```diff
@@ -0,0 +1,339 @@
...
+
+Google Analytics 4 has some reserved events names which cannot be used. Refer to the [Reserved names](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#reserved_names) for complete list of reserved event names.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:334]:
```suggestion
Google Analytics 4 has some reserved event, parameter, and user property names that cannot be used. Refer to the [Measurement Protocol (Google Analytics 4)](https://developers.google.com/analytics/devguides/collection/protocol/ga4)](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#reserved_names) for complete list of reserved event names.
```

File: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 20]
```diff
@@ -0,0 +1,339 @@
...
+- firebase_
+- ga_
+- google_
+- gtag.
+
+Refer to the [GA4 document](https://support.google.com/analytics/answer/10085872#zippy=%2Cweb%2Cin-this-article%2Creserved-prefixes-and-event-names) for more information.
+
+### What are the reserved event, parameter, and user property names in Google Analytics 4?
+
+Google Analytics 4 has some reserved event, parameter, and user property names that cannot be used. Refer to the [Measurement Protocol (Google Analytics 4)](https://developers.google.com/analytics/devguides/collection/protocol/ga4)](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#reserved_names) guide for a complete list of reserved names.
```
Comment (akashrpo) [inline destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx:334]:
```suggestion
Google Analytics 4 has some reserved event, parameter, and user property names that cannot be used. Refer to the [Measurement Protocol (Google Analytics 4)](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#reserved_names) guide for a complete list of reserved names.
```

**Review by akashrpo — APPROVED** [review]
LGTM

**Review by shrouti1507 — APPROVED** [review]
LGTM

**Pull request merged**